### PR TITLE
Core: Testing Module UI

### LINKED ISSUE #29096: Testing module UI
UI component with the basic elements:

- Floating container at the bottom of the sidebar
- List each registered test provider with their icon, title and description (`render`)
- Add a play button for each test provider, and let those trigger events as specified in https://github.com/storybookjs/storybook/issues/29093
- Show stop button when running
- Handle disabled state (i.e. not runnable / cancelable)
- Show filter buttons for errors/warnings

Demo here: https://storybook-tests.vercel.app/test-3

### PR DESCRIPTION
**Pull request by ghengeveld**
Closes #29096

## What I did

This implements the UI for the Testing Module. It sits docked in the sidebar, floating at the bottom. Notifications were updated to sit in the sidebar as well, just above the Testing Module. The sidebar has a `div#sidebar-bottom` element with some CSS variables attached to allow integrators such as Chromatic to add custom elements here, whilst retaining a consistent UI.

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  77.7 MB | 77.7 MB | 0 B | - | 0% |
| initSize |  152 MB | 152 MB | 19.1 kB | **Infinity** | 0% |
| diffSize |  74.4 MB | 74.5 MB | 19.1 kB | **Infinity** | 0% |
| buildSize |  7.27 MB | 7.28 MB | 12.3 kB | **Infinity** | 0.2% |
| buildSbAddonsSize |  1.87 MB | 1.87 MB | 651 B | **Infinity** | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  1.92 MB | 1.93 MB | 9.47 kB | **Infinity** | 0.5% |
| buildSbPreviewSize |  312 kB | 312 kB | 0 B | - | 0% |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  4.3 MB | 4.31 MB | 10.1 kB | **Infinity** | 0.2% |
| buildPreviewSize |  2.97 MB | 2.98 MB | 2.14 kB | **Infinity** | 0.1% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  5.5s | 21s | 15.4s | **Infinity** | 🔺73.5% |
| generateTime |  19.6s | 19.3s | -303ms | **-Infinity** | -1.6% |
| initTime |  14.4s | 14s | -418ms | **-Infinity** | -3% |
| buildTime |  13s | 9.5s | -3s -537ms | **-Infinity** | 🔰-37.1% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  8.4s | 6.3s | -2s -128ms | **-Infinity** | 🔰-33.7% |
| devManagerResponsive |  5.4s | 4s | -1s -324ms | **-Infinity** | 🔰-32.4% |
| devManagerHeaderVisible |  719ms | 591ms | -128ms | **-Infinity** | 🔰-21.7% |
| devManagerIndexVisible |  758ms | 638ms | -120ms | **-Infinity** | 🔰-18.8% |
| devStoryVisibleUncached |  1.6s | 713ms | -897ms | **-Infinity** | 🔰-125.8% |
| devStoryVisible |  756ms | 639ms | -117ms | **-Infinity** | 🔰-18.3% |
| devAutodocsVisible |  626ms | 570ms | -56ms | **-Infinity** | 🔰-9.8% |
| devMDXVisible |  655ms | 596ms | -59ms | **-Infinity** | 🔰-9.9% |
| buildManagerHeaderVisible |  574ms | 528ms | -46ms | **-Infinity** | 🔰-8.7% |
| buildManagerIndexVisible |  658ms | 594ms | -64ms | **-Infinity** | 🔰-10.8% |
| buildStoryVisible |  659ms | 596ms | -63ms | **-Infinity** | 🔰-10.6% |
| buildAutodocsVisible |  645ms | 483ms | -162ms | **-Infinity** | 🔰-33.5% |
| buildMDXVisible |  550ms | 491ms | -59ms | **-Infinity** | 🔰-12% |

## Greptile Summary

This PR implements the UI for the Testing Module in Storybook, focusing on creating a floating container at the bottom of the sidebar with test provider information and controls.

- Added new TestingModule component in `code/core/src/manager/components/sidebar/TestingModule.tsx`
- Updated SidebarBottom component to include TestingModule and notifications
- Implemented collapsible UI with run/stop buttons and error/warning filters
- Added 'runnable' and 'watchable' properties to test provider configuration in `code/addons/test/src/manager.tsx`
- Updated NotificationList and NotificationItem components for improved responsiveness and styling

### CODE REVIEW

**Review by yannbf — CHANGES_REQUESTED** [review]
Awesome stuff!!

Here's some testing feedback we should fix before merging (if applicable, else we should create separate tickets for them):

### Watch mode state is not shown upon click
As you can see, the button won't change into an active/inactive state. Although watch mode toggling works, it doesn't feel like it in the UI

**Expected behavior:** The `eye` icon should be switched to active/inactive state upon state change.


### Loading state is not shown on test run
Although there is a story for the loading state with a spinner and where the icon shows a square, clicking on the run button doesn't trigger that
**Expected behavior:** The spinner should shown upon test runs, as well as the stop icon.


### Run all tests does not run tests
It does spawn vitest, but no tests are run
**Expected behavior:** All tests should be triggered, as well as the loading state of the components should be toggled.


### Error/warning filter colors are not applied

**Expected behavior:** Colors should be applied correctly

image

That's because there are other colors overwriting the ones defined in the component, this happens as you set the Button to have a `ghost` variant in [this line of code](https://github.com/storybookjs/storybook/blob/21d01be508e9c9ec9c4d2a106c1f6384fae92122/code/core/src/manager/components/sidebar/TestingModule.tsx#L281):



### The testing module is available on Mobile

Although it looks relatively good (though the shadow seems excessive on dark mode), I thought the testing module should not appear at all on mobile?

**Expected behavior:** Testing module should not be rendered on mobile (unless decided otherwise)



It also clashes with the notification element:


### Notifications look off on mobile (dark mode)

Because the notifications appear on any view on mobile, when in dark mode they have a weird shadow that only makes sense if they're rendered on the sidebar:

File: code/core/src/manager/components/sidebar/TestingModule.tsx [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+        >
+          <Content ref={contentRef}>
+            {testProviders.map(
+              ({ id, icon, title, description, runnable, running, watchable, watching }) => (
+                <TestProvider key={id}>
+                  <Info>
+                    <Icon>{icon}</Icon>
+                    <Details>
+                      <Title>{title}</Title>
+                      <Description>{description?.({})}</Description>
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply code/core/src/manager/components/sidebar/TestingModule.tsx:228]: Description is actually required, so should never be null.

File: code/core/src/manager/components/sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -200,25 +177,12 @@ export const Sidebar = React.memo(function Sidebar({
             )}
           </Search>
         </Top>
+        {isLoading ? null : (
+          <div className="sb-bar">
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply code/core/src/manager/components/sidebar/Sidebar.tsx:181]: I think this is legacy. Not sure where it's used but I just left it there.

File: code/core/src/manager/components/sidebar/TestingModule.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+          }[status],
+          color: {
+            negative: theme.color.negativeText,
+            warning: theme.color.warningText,
+          }[status],
+        }
+      : {
+          background: {
+            negative: `${theme.color.negative}22`,
+            warning: `${theme.color.warning}22`,
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply code/core/src/manager/components/sidebar/TestingModule.tsx:120]: Yes this is transparency. Used in a bunch of places, doesn't seem worthy of a comment tbh.

**Comment by ghengeveld** [comment]
**Watch mode state is not shown upon click**
We'll handle this when we deal with status updates in a later PR.

**Loading state is not shown on test run**
Same.

**Run all tests does not run tests**
Fixed.

**Error/warning filter colors are not applied**
Fixed.

**The testing module is available on Mobile**
Fixed.

**Notifications look off on mobile (dark mode)**
Fixed.

File: code/core/src/manager/components/sidebar/TestingModule.tsx [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+  setErrorsActive,
+  warningCount,
+  warningsActive,
+  setWarningsActive,
+  onRunTests,
+  onSetWatchMode,
+}: TestingModuleProps) => {
+  const contentRef = useRef<HTMLDivElement>(null);
+  const [collapsed, setCollapsed] = useState(false);
+  const [maxHeight, setMaxHeight] = useState(500);
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply code/core/src/manager/components/sidebar/TestingModule.tsx:189]: Done

**Review by yannbf — APPROVED** [review]

**Pull request merged**